XenAdmin/SettingsPanels/USBEditPage.cs

```diff
@@ -200,7 +223,29 @@ private void buttonAttach_Click(object sender, System.EventArgs e)
 
         private void buttonDetach_Click(object sender, System.EventArgs e)
         {
-
+            if ((selectedRow != null) && (_vm != null))
+            {
+                bool confirmed = false;
+                using (var dlg = new ThreeButtonDialog(
+                    new ThreeButtonDialog.Details(null, Messages.ACTION_VUSB_DETACH_CONFIRM, Messages.ACTION_VUSB_DETACH),
+                    ThreeButtonDialog.ButtonYes,
+                    new ThreeButtonDialog.TBDButton(Messages.NO_BUTTON_CAPTION, DialogResult.No, ThreeButtonDialog.ButtonType.CANCEL, true)))
+                {
+                    if (dlg.ShowDialog(Program.MainWindow) == DialogResult.Yes)
+                    {
+                        confirmed = true;
+                    }
+                }
+                if (confirmed)
+                {
+                    // Run this stuff off the event thread, since it involves a server call
+                    System.Threading.ThreadPool.QueueUserWorkItem((System.Threading.WaitCallback)delegate (object o)

```

Comment:
Teh extra thread is not needed here, as the action runs on a separate thread already.